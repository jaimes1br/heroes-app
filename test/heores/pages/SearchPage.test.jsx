import { fireEvent, render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { SearchPage } from "../../../src/heores";

const mockUseNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockUseNavigate
}));


describe('test on <SearchPage/>', () => {
  
    beforeEach(() => jest.clearAllMocks() );

    test('should render with default values', () => {
        const { container } = render (
            <MemoryRouter>
                <SearchPage/>                
            </MemoryRouter>
        );

        expect( container ).toMatchSnapshot();
    });

    test('should render spiderman witg the value in queryString', () => {
        
        render (
            <MemoryRouter initialEntries={['/search?q=spider%20man']}>
                <SearchPage/>                
            </MemoryRouter>
        );
        
        const input = screen.getByRole('textbox');
        expect( input.value ).toBe('spider man');
        
        const img = screen.getByRole('img');
        expect( img.src ).toContain('/assets/heroes/marvel-spider.jpg')
    });
    

    test('should show error if hero(bob esponja) is not found', () => {
      
        render (
            <MemoryRouter initialEntries={['/search?q=bob esponja']}>
                <SearchPage/>                
            </MemoryRouter>
        );

        expect( screen.getByLabelText('errorText')).toBeTruthy();
        
    });

    test('should call navigate to the new page', () => {
      
        const inputValue = 'spider';

        render (
            <MemoryRouter initialEntries={['/search']}>
                <SearchPage/>                
            </MemoryRouter>
        );

        const input = screen.getByRole('textbox');
        fireEvent.change(input, { target: { name: 'searchText', value: inputValue }});

        const form = screen.getByRole('form');
        fireEvent.submit( form );

        expect( mockUseNavigate ).toHaveBeenCalledWith(`?q=${inputValue}`);
    });
    
    
});
