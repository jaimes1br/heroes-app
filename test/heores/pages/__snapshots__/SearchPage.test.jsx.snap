// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <SearchPage/> should render with default values 1`] = `
<div>
  <h1
    class="mt-2"
  >
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form
        aria-label="form"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-dark mt-1"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="alert alert-primary"
      >
         Search a hero 
      </div>
    </div>
  </div>
</div>
`;
